core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -1131,6 +1337,230 @@ TEST(TEST_CATEGORY, mathematical_functions_absolute_value) {
+    ASSERT_EQ(errors, 0);
+  }
+  KOKKOS_FUNCTION void operator()(int, int& e) const {
+    using Kokkos::remainder;
+    if (!compare(remainder(6.2f, 4.f), 2.2f, 2) &&
+        !compare(remainder(-6.2f, 4.f), 2.2f, 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed remainder(float)\n");
+    }
+    if (!compare(remainder(static_cast<KE::half_t>(6.2f),
+                           static_cast<KE::half_t>(4.f)),
+                 static_cast<KE::half_t>(2.2f), 1) &&
+        !compare(remainder(static_cast<KE::half_t>(-6.2f),
+                           static_cast<KE::half_t>(4.f)),
+                 -static_cast<KE::half_t>(2.2f), 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed remainder(KE::half_t)\n");
+    }
+    if (!compare(remainder(static_cast<KE::bhalf_t>(6.2f),
+                           static_cast<KE::bhalf_t>(4.f)),
+                 static_cast<KE::bhalf_t>(2.2f), 1) &&
+        !compare(remainder(static_cast<KE::bhalf_t>(-6.2f),
+                           static_cast<KE::bhalf_t>(4.f)),
+                 -static_cast<KE::bhalf_t>(2.2f), 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed remainder(KE::bhalf_t)\n");
+    }
+    if (!compare(remainder(6.2, 4.), 2.2, 2) &&
+        !compare(remainder(-6.2, 4.), 2.2, 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed remainder(double)\n");
+    }
+#ifdef MATHEMATICAL_FUNCTIONS_HAVE_LONG_DOUBLE_OVERLOADS
+    if (!compare(remainder(6.2l, 4.l), 2.2l, 1) &&
+        !compare(remainder(-6.2l, 4.l), -2.2l, 1)) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF("failed remainder(long double)\n");
+    }
+#endif
+    // special values
+    using Kokkos::isinf;
+    using Kokkos::isnan;
+    if (!isinf(remainder(-KE::infinity<float>::value, 1.f)) &&
+        !isnan(remainder(-KE::quiet_NaN<float>::value, 1.f))) {
+      ++e;
+      KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+          "failed remainder(floating_point) special values\n");
+    }
+
+    static_assert(
+        std::is_same<decltype(remainder(static_cast<KE::half_t>(4.f),
+                                        static_cast<KE::half_t>(4.f))),
+                     KE::half_t>::value,
+        "");
+    static_assert(
+        std::is_same<decltype(remainder(static_cast<KE::bhalf_t>(4.f),
+                                        static_cast<KE::bhalf_t>(4.f))),
+                     KE::bhalf_t>::value,
+        "");
+    static_assert(std::is_same<decltype(remainder(4.f, 4.f)), float>::value,
+                  "");
+    static_assert(std::is_same<decltype(remainder(5., 5.)), double>::value, "");
+#ifdef MATHEMATICAL_FUNCTIONS_HAVE_LONG_DOUBLE_OVERLOADS
+    static_assert(
+        std::is_same<decltype(remainder(6.l, 6.l)), long double>::value, "");
+#endif
+  }
+};
+
+TEST(TEST_CATEGORY, mathematical_functions_ieee_remainder_function) {
+  TestIEEEFloatingPointRemainderFunction<TEST_EXECSPACE>();
+}
+#endif
+
+// TestFpClassify
+
+// TestIsFinite
+
+// TestIsInf
```

Comment:
What are these?  A TODO?

Reply:
Not unless @srajama1 or @lucbv requests it.

Reply:
https://github.com/kokkos/kokkos/issues/6279